JSDemos/Demos/TabPanel/SortableClosableTabs/React/App.js

```diff
@@ -1,119 +1,109 @@
-﻿import React from 'react';
+﻿import React, { useState } from 'react';
 import { Button } from 'devextreme-react';
 import { Sortable } from 'devextreme-react/sortable';
 import TabPanel from 'devextreme-react/tab-panel';
 import 'devextreme/data/odata/store';
 
 import service from './data.js';
 import EmployeeTemplate from './EmployeeTemplate.js';
-import { CloseButton } from './CloseButton.js';
 
 const allEmployees = service.getEmployees();
 
-class App extends React.Component {
-  constructor() {
-    super();
+function App() {
+  const [employees, setEmployees] = useState(allEmployees.slice(0, 3));
+  const [selectedItem, setSelectedItem] = useState(allEmployees[0]);
 
-    this.state = {
-      employees: allEmployees.slice(0, 3),
-      selectedItem: allEmployees[0]
-    };
-  }
+  function addButtonHandler() {
+    const newItem = allEmployees
+      .filter((employee) => employees.indexOf(employee) === -1)[0];
 
-  render() {
-    return (
-      <div>
-        <div id="container">
-          <Button
-            disabled={this.disableButton()}
-            text="Add Tab"
-            icon="add"
-            type="default"
-            onClick={this.addButtonHandler}
-          />
-        </div>
-        <Sortable
-          filter=".dx-tab"
-          data={this.state.employees}
-          itemOrientation ="horizontal"
-          dragDirection="horizontal"
-          onDragStart={this.onTabDragStart}
-          onReorder={this.onTabDrop}
-        >
-          <TabPanel
-            dataSource={this.state.employees}
-            height={472}
-            itemTitleRender={this.renderTitle}
-            deferRendering={false}
-            showNavButtons={true}
-            selectedItem={this.state.selectedItem}
-            repaintChangesOnly={true}
-            onSelectionChanged={this.onSelectionChanged}
-            itemComponent={EmployeeTemplate}>
-          </TabPanel>
-        </Sortable>
-      </div>
-    );
+    setEmployees([...employees, newItem]);
+    setSelectedItem(newItem);
   }
 
-  onTabDragStart = (e) => {
-    e.itemData = e.fromData[e.fromIndex];
+  function needDisableAddButton() {
+    return employees.length === allEmployees.length;
   }
 
-  onTabDrop = (e) => {
-    const employees = [...this.state.employees];
-    const selectedItem = employees[e.fromIndex];
+  function closeButtonHandler() {
+    const newEmployees = [...employees];
+    const index = newEmployees.indexOf(selectedItem);
 
-    employees.splice(e.fromIndex, 1);
-    employees.splice(e.toIndex, 0, e.itemData);
+    newEmployees.splice(index, 1);
+    setEmployees(newEmployees);
 
-    this.setState({
-      selectedItem,
-      employees,
-     
-    });
+    if (index >= newEmployees.length && index > 0) {
```

Comment:
Do we need this condition? There were issues and index can be '-1'?

Reply:
It looks like this is a redundant check. It does nothing in case of negative index or similar bound cases